src/views/PositionPlot/PositionPlotView.tsx

```diff
@@ -90,30 +114,43 @@ const PositionPlotView: FunctionComponent<Props> = ({data, width, height}) => {
         return {yMin, yMax}
     }, [series])
 
+    const pixelTransform = use2dPanelDataToPixelMatrix(
+        pixelsPerSecond,
+        visibleTimeStartSeconds,
+        valueRange.yMin,
+        valueRange.yMax,
+        1,
+        panelHeight,
+        true
+    )
+
     const panels: TimeScrollViewPanel<PanelProps>[] = useMemo(() => {
-        const yMap = ((y: number) => (
-            (1 - (y - valueRange.yMin) / (valueRange.yMax - valueRange.yMin)) * panelHeight
-        ))
+        const pixelZero = getYAxisPixelZero(pixelTransform)
+        // this could also be done as one matrix multiplication by concatenating the dimensions;
+        // and we could even separate out the time series values (which are repeated). But probably not worth it.
+        // TODO: ought to profile these two versions
+        const pixelData = series.map(s => {
+            const augmentedPoints = matrix([ s.times, s.values, new Array(s.times.length).fill(1) ])
+            const pixelPoints = multiply(pixelTransform, augmentedPoints).valueOf() as number[][]
```

Comment:
This uses the 2d data-to-pixel transformation matrix provided by `TimeSeriesView.use2dPanelDataToPixelMatrix()` (called above) to vectorize the conversion from dataspace to pixelspace. Technically we could reduce the amount of computation by a) reshaping the data so the dimensions are all done at once, and b) operating on the (repeated) time elements once, and then converting only the y-axis data, but a) seems like more trouble than it's worth for small numbers of dimensions and b) would require another conversion-matrix hook plus chopping up the data more, so I've left it alone for now. (I give myself permission by noting that the existing version is also converting the times into pixels once per dimension.)